client/payments/data/actions.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { apiFetch } from '@wordpress/data-controls';
+import { dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import TYPES from './action-types';
+import { API_NAMESPACE } from './constants';
+
+export function handleFetchError( error, message ) {
+	const { createNotice } = dispatch( 'core/notices' );
```

Comment:
Sorry to say you can't really do this. well it might "work" when both stores are together but its an anti pattern. What you need to do instead is call createNotice from the UI, you're not supposed to dispatch actions to other stores from within a resolver.

Also when you do `yield handleFetchError` you're yielding undefined, so it'll do nothing useful. In some of our other stores we store an error state in the reducer instead, what you could do, is write a selector to get the response but that also returns any error state if that exists. Then in the component if the error state is on the response you can do a `createNotice` there.

Reply:
Also worth mentioning why its so bad to do createNotice in a store: these are UI elements that will be seen by users, if we want to change translated copy for them, that belongs at the UI layer.

Reply:
Cool, I will look up other stores and see how they handle it.